examples/7_extended_write_serial.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+"""
+This file is part of the openPMD-api.
+
+Copyright 2018 openPMD contributors
+Authors: Axel Huebl, Fabian Koller
+License: LGPLv3+
+"""
+from openPMD import Series, Access_Type, Dataset, Mesh_Record_Component, \
+    Unit_Dimension, Datatype, determine_datatype
+import numpy as np
+
+
+SCALAR = Mesh_Record_Component.SCALAR
+
+
+if __name__ == "__main__":
+    # open file for writing
+    f = Series(
+        "working/directory/2D_simData_py.h5",
+        Access_Type.create
+    )
+
+    # all required openPMD attributes will be set to reasonable default values
+    # (all ones, all zeros, empty strings,...)
+    # manually setting them enforces the openPMD standard
+    f.set_meshes_path("custom_meshes_path")
+    f.set_particles_path("long_and_very_custom_particles_path")
+
+    # it is possible to add and remove attributes
+    f.set_comment("This is fine and actually encouraged by the standard")
+    f.set_attribute(
+        "custom_attribute_name",
+        "This attribute is manually added and can contain about any datatype "
+        "you would want"
+    )
+    # note that removing attributes required by the standard typically makes
+    # the file unusable for post-processing
+    f.delete_attribute("custom_attribute_name")
+
+    # setting attributes can be chained in JS-like syntax for compact code
+    tmpItObj = f.iterations[1] \
+        .set_time(42.0) \
+        .set_dt(1.0) \
+        .set_time_unit_SI(1.39e-16)
+    # everything that is accessed with [] should be interpreted as permanent
+    # storage the objects sunk into these locations are deep copies
+    del tmpItObj
+    f.iterations[2].set_comment("This iteration will not appear in any output")
+    del f.iterations[2]
+
+    # this is a reference to an iteration
+    reference = f.iterations[1]
+    reference.set_comment("Modifications to a reference will always be visible"
+                          " in the output")
+    del reference
+
+    # alternatively, a copy may be created and later re-assigned to
+    # f.iterations[1]
+    copy = f.iterations[1]  # TODO .copy()
+    copy.set_comment("Modifications to copies will only take effect after you "
+                     "reassign the copy")
+    f.iterations[1] = copy
+    del copy
+
+    f.iterations[1].delete_attribute("comment")
+
+    cur_it = f.iterations[1]
+
+    # the underlying concept for numeric data is the openPMD Record
+    # https://github.com/openPMD/openPMD-standard/blob/upcoming-1.0.1/STANDARD.md#scalar-vector-and-tensor-records
+    # Meshes are specialized records
+    cur_it.meshes["generic_2D_field"].set_unit_dimension(
+        {Unit_Dimension.L: -3, Unit_Dimension.M: 1})
+
+    # as this is a reference, it modifies the original resource
+    lowRez = cur_it.meshes["generic_2D_field"]
+    lowRez.set_grid_spacing([1, 1, 1, 1, 1, 1]) \
```

Comment:
This `gridSpacing` would actually make it a 6D field.

Reply:
huh, don't we have the same in the C++ example? Should be three values in both examples, no?

Reply:
Oh wow, that's dangerous xD
I thought the initializer-list constructor should read `({6, 1})` but your code is indeed ok and does exactly what you say.